pkgs/by-name/sm/smartcat/package.nix

```
@@ -0,0 +1,41 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+
+, openssl
+, pkg-config
+, libX11
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "smartcat";
+  version = "1.3.0";
+
+  src = fetchFromGitHub {
+    owner = "efugier";
+    repo = "smartcat";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-iCtNNKXo0peGGUaQXKaaYaEo7MAL70PX0BAWPERNmlo=";
+  };
+
+  cargoLock.lockFile = "${src}/Cargo.lock";
```

```suggestion
  cargoHash = "sha256-qNl2bI5VqpMfoFc+gZC4XHrNT9pnWseduYSOi5Dzr9M=";
```
ig CI is failing due to an IFD

---

Ah, I read the section about that in the contribution guide but didn't realize that would count. Will fix ASAP.

---

The lockfile is read into the nix evaluator ("import"):

https://github.com/NixOS/nixpkgs/blob/5a8c65e118c36697efb1fcec3093d915dd68a8ff/pkgs/build-support/rust/import-cargo-lock.nix#L43-L46

which depends on a derivation (`src`), which is why this ends up being IFD

---

Right, that makes a lot of sense in hindsight. Probably safe to assume that's a given any time you pass a path as an argument to a builder, I'll keep it in mind.

---

> Probably safe to assume that's a given any time you pass a path as an argument to a builder

Not necessarily - but lockfiles are a special case, because those dependencies need to be fetched in advance, so nix needs to read the lockfile to know what to fetch. And there are some language ecosystems where IFD is permitted (haskell and R?) so it's case-by-case really

---

I see, thanks for elaborating